pkg/scheduler/framework/plugins/podtopologyspread/scoring.go

```diff
@@ -168,26 +182,26 @@ func (pl *PodTopologySpread) Score(ctx context.Context, cycleState *framework.Cy
 	}
 
 	// Return if the node is not qualified.
-	if _, ok := s.NodeNameSet[node.Name]; !ok {
+	if s.IgnoredNodes.Has(node.Name) {
 		return 0, nil
 	}
 
 	// For each present <pair>, current node gets a credit of <matchSum>.
 	// And we sum up <matchSum> and return it as this node's score.
-	var score int64
-	for _, c := range s.Constraints {
+	var score float64
+	for i, c := range s.Constraints {
 		if tpVal, ok := node.Labels[c.TopologyKey]; ok {
+			var cnt int64
 			if c.TopologyKey == v1.LabelHostname {
-				count := countPodsMatchSelector(nodeInfo.Pods, c.Selector, pod.Namespace)
-				score += int64(count)
+				cnt = int64(countPodsMatchSelector(nodeInfo.Pods, c.Selector, pod.Namespace))
 			} else {
 				pair := topologyPair{key: c.TopologyKey, value: tpVal}
-				matchSum := *s.TopologyPairToPodCounts[pair]
-				score += matchSum
+				cnt = *s.TopologyPairToPodCounts[pair]
 			}
+			score += float64(cnt) * s.TopologyWeight[i]
```

Comment:
Lets have a comment clarifying what the weight is trying to do, because it is not really trying to give preference to one topology over the other, it is more like a factor to make sure smaller topologies don't get watered down because of the way we calculate the score.

Reply:
Done